### ISSUE TITLE
lilliput causes crashes when minuit output is printed (Trac #1842)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1834]
When I3GulliverMinuit prints minuit output (USE_ROOT=ON), this causes everything to crash and burn. Using trunk with lilliput revision 146654 - lilliput/gulliver is being called through millipede here. Could this be related to the (apparently) missing error matrix?

INFO (lilliput):  FCN=215.73 FROM SIMPLEX   STATUS=PROGRESS        3 CALLS           4 TOTAL (TMinuit.cxx:5844 in virtual void lilliput::TMinuit::mnprin(int32_t, double))
INFO (lilliput):                      EDM=5.46236    STRATEGY= 2      NO ERROR MATRIX        (TMinuit.cxx:5849 in virtual void lilliput::TMinuit::mnprin(int32_t, double))
ERROR (I3Module): MonopodFit_DR: Exception thrown (I3Module.cxx:116 in void I3Module::Do(void (I3Module::*)()))
IndexError: basic_string::substr
INFO (I3Gulliver): (MonopodFit_DR) nr. of proper llh function calls by minimizer: 0 (I3Gulliver.cxx:32 in virtual I3Gulliver::~I3Gulliver())

Migrated from https://code.icecube.wisc.edu/ticket/1842
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:18", 
    "description": "When I3GulliverMinuit prints minuit output (USE_ROOT=ON), this causes everything to crash and burn. Using trunk with lilliput revision 146654 - lilliput/gulliver is being called through millipede here. Could this be related to the (apparently) missing error matrix?\n\nINFO (lilliput):  FCN=215.73 FROM SIMPLEX   STATUS=PROGRESS        3 CALLS           4 TOTAL (TMinuit.cxx:5844 in virtual void lilliput::TMinuit::mnprin(int32_t, double))\nINFO (lilliput):                      EDM=5.46236    STRATEGY= 2      NO ERROR MATRIX        (TMinuit.cxx:5849 in virtual void lilliput::TMinuit::mnprin(int32_t, double))\nERROR (I3Module): MonopodFit_DR: Exception thrown (I3Module.cxx:116 in void I3Module::Do(void (I3Module::*)()))\nIndexError: basic_string::substr\nINFO (I3Gulliver): (MonopodFit_DR) nr. of proper llh function calls by minimizer: 0 (I3Gulliver.cxx:32 in virtual I3Gulliver::~I3Gulliver())", 
    "reporter": "snowicki", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067198646291", 
    "component": "combo reconstruction", 
    "summary": "lilliput causes crashes when minuit output is printed", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-08-30T20:45:25", 
    "milestone": "", 
    "owner": "kkrings", 
    "type": "rumor / allegation"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160831T08:55:30: boersma commented:

> Could you point me to a script that I can use to reproduce this error? On which platform are you running? If not on cobalt, then could you check that you get the same error on cobalt?
> 
> I did a grep in all of icerec/trunk and got zero hits for `MonopodFit_DR`. The `simplefit` test in `gulliver-modules` (which uses `TMinuit`) runs without problems. Note that unless you use a release from more than about a year old, you do not actually need USE_ROOT=ON in order to use `TMinuit`, because as you can see from the INFO output, `lilliput` uses its own copy of `TMinuit`. Also, in order to build icerec/trunk on the cobalts I needed the `py2-v3_early_access` setup from `cvmfs`, which comes without any ROOT, so you need to hack that somehow into your environment if you really really want to (`cmake` silently ignored my `-DUSE_ROOT=ON` instruction).

**Comment by icecube-trac** [comment]
Trac update at 20160831T12:47:10: 
* kkrings changed status from "new" to "accepted"
* kkrings changed milestone from "" to "September 2016"
* kkrings changed type from "defect" to "rumor / allegation"
* kkrings changed owner from "" to "kkrings"

**Comment by icecube-trac** [comment]
Trac update at 20160905T08:26:50: 
* kkrings commented:

> In changeset 149804:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149804"
> Fix bug in the local TMinuit copy that was introduced when replacing ROOT classes with STL and Boost equivalents: std::string::substr raises out_of_range exception when the position is larger than the string length while TString simply returns an empty string. This fix closes #1834.
> ```


* kkrings changed resolution from "" to "fixed"
* kkrings changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:18: 
* olivas commented:

> Milestone September 2016 deleted


* olivas changed milestone from "September 2016" to ""
